blogs/cli/index.md

```diff
@@ -0,0 +1,442 @@
+```
+SSLKEYLOGFILE="keylog.txt" chromium-browser &
+```
+
+This generated file contains each client and server traffic secrets to decode your packets.
+
+Then in Wireshark, specify the file path in `Edit→Preferences→Protocols→TLS→(Pre)-Master-Secret log filename`. A single file can hold keys for multiple TLS connections.
+
+![wireshark tls keylog](./images/wireshark_tls_keylog.png)
+
+You should now have access to decrypted packets !
+
+What about a secure connection between two pods ? 
+For these cases, you need to have a good understanding of what's involved behind the scene. If your application rely on OpenSSL, GnuTLS or NSS TLS library, you should be able to set the `SSLKEYLOGFILE` enviroment variable and extract the result.
+
+For our nodejs example, you need to either use the cli option [--tls-keylog](https://nodejs.org/api/cli.html#cli_tls_keylog_file) or modify the code and subscribe on [keylog event](https://nodejs.org/api/tls.html#event-keylog). When multiple teams are involved, that may take a while !
+
+Another alternative is to get a man in the middle approach using a proxy. This can be done using [kubetap](https://soluble-ai.github.io/kubetap/).
+
+You can install it using krew by running:
+```
+oc krew install tap
+```
+
+Then you can run the following command to port forward the traffic of your service to the proxy:
+```
+oc tap on -n packet-capture-scenario -p 3001 --https nodejs-basic --port-forward
+```
+
+The output should look like this:
+```
+Establishing port-forward tunnels to Service...
+                                                                                     
+
+Port-Forwards:
+
+  mitmproxy - http://127.0.0.1:2244
+  nodejs-basic - http://127.0.0.1:4000
+```
+
+Open the mitm proxy url and generate a query to your route. You will see queries listed in the proxy web UI with the content displayed:
+
+![mitm proxy](./images/mitm_proxy.png)
+
+## Future improvements
+
+For now, the Network Observability CLI is only providing `capture commands` such as [many other observability projects](https://github.com/collabnix/kubetools?tab=readme-ov-file#observability) but it will improve to embed tools to facilitate your life as much as possible.
+
+We have in mind to improve the capture process by automatically finding ips and ports matching to the pods you are looking for and automatically filter on these.
+
+The packet capture needs to be improved to align with flows filter capabilities and facilitate decryption of traffic. We are also investigating on ways to embed enriched information inside the pcap file format.
+
+Also, the CLI could hold additionnal functionnalities such as, installing and configuring the Network Observability Operator, in a step by step setup way, including Loki and Prometheus configurations.
```

Comment:
```suggestion
Also, the CLI could hold additional functionalities, such as installing and configuring the Network Observability Operator in a step-by-step setup, including Loki and Prometheus configurations.
```

Reply:
Some frenglish translation and comma rearranging 🙃

Reply:
![image](https://github.com/user-attachments/assets/a29df2c0-b5a4-41b1-8a2a-5f980d72efcb)